### PR TITLE
Prefer deploy to deployAndVerify for Vyper contracts

### PR DESCRIPTION
**Pull request by nventuro**
This also makes `deploy` much more similar to `deployAndVerify`.

### CODE REVIEW

File: pkg/deployments/src/task.ts [hunk 1]
```diff
@@ -91,19 +91,19 @@ export default class Task {
       return await this.check(name, args, libs);
```
Comment (TomAFrench) [inline pkg/deployments/src/task.ts:91]: This check on `TaskMode.CHECK` is redundant as we do another one right inside `deploy`. We *could* remove it if we're confident that we're not going to have any logic happening before the deploy in `deployAndVerify` (which seems reasonable).

**Review by TomAFrench — APPROVED** [review]
Looks good. We're double checking for if we're in `CHECK` mode but you could argue that's a safety check for if we modify `deployAndVerify` in future.

File: pkg/deployments/src/task.ts (hunk 1 above)
Comment (nventuro) [reply pkg/deployments/src/task.ts:91]: Yes, but that'd cause for CHECK executions to attempt verification, which we don't want to.

Comment (TomAFrench) [reply pkg/deployments/src/task.ts:91]: Ah true, I forgot how functions work for a bit.

**Pull request merged**